loader/src/bootconfig.rs

```diff
@@ -51,6 +53,30 @@ pub struct BootConfig {
 
     /// The number of 'Init' tags discovered
     pub init_process_count: usize,
+
+    /// Amount that init_size is offset by swap. We have to track this
+    /// separately because init_size is used during allocations to track
+    /// cfg_top(), but then re-used during page mapping with the assumption
+    /// that it also points to exclusive kernel memory. swap_offset allows
+    /// us to subtract out the memory we allocated and gave to swap in that
+    /// phase of boot. When swap is not enabled, it is set to 0.
+    pub swap_offset: usize,
```

Comment:
Should this be gated by `#[cfg(feature = "swap")]` as well

Reply:
It's not gated in `swap` because I made an enum that handles `IniS` types that's not gated by swap. The trade-off is it clutters the code quite a bit to gate all the features, instead what I do is leave the `IniS` cases empty in the case that swap is not enabled, for what I think is a fairly minor cost in binary size.

Reply:
For example, this code:

https://github.com/betrusted-io/xous-core/blob/336e47eabe44e6153bfd00a967c09b7ecb856f8c/loader/src/main.rs#L552-L557

To gate out the IniS branch, you'd have to put the feature wrap around the entire if/else (I don't think you can gate just an else clause), and that's kind of gross. So I leave some minimal stubs to detect an IniS tag and throw errors if one is encountered when it shouldn't be (which I think is probably not a bad thing anyways); and in the "normal" case `swap_offset` just remains at 0 and doesn't cause any troubles. It's also possibly more generically useful to track offsets in the future in case we need to allocate pages to hand back to processes that aren't swap-related.